test/integration/default/package_spec.rb

```diff
@@ -28,3 +28,31 @@
 describe package('nginx') do
   it { should_not be_installed }
 end
+
+# the following test will iterate over all packages retrieved via WMI and verifies that the
+# optimized package implementation is returning the same results
+if os.windows?
+  # compare optimized version with wmi results
+  packages = powershell("Get-WmiObject -Class Win32_Product |  % { $_.Name }").stdout.strip.split("\n")
+
+  packages.each { |pkg|
+    package_name = pkg.strip
+    # get wmi package information
+    cmd = powershell("Get-WmiObject -Class Win32_Product | Where-Object {$_.Name -eq '#{package_name}'} | Select-Object -Property Name,Version,Vendor,PackageCode,Caption,Description | ConvertTo-Json")
```

Comment:
👍 I like that we are using the old command here to ensure we match the same output.